### PR TITLE
Create favourite and date filter

### PR DESCRIPTION
**Pull request by Jiay0928**
Screen Shot 15
Screen Shot 16

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+								.invokeOnCompletion {
+									if (!datePickerBottomSheetState.isVisible) {
+										openDatePicker = false
+									}
+									startButtonText =
+										if (dateRangePickerState.selectedStartDateMillis != null) {
+											displayDateFormatter.formatDate(
+												dateRangePickerState.selectedStartDateMillis,
+												locale = Locale.getDefault()
+											)!!
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Just do `?: ""` instead

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+										} else {
+											""
+
+										}
+									endButtonText =
+										if (dateRangePickerState.selectedEndDateMillis != null) {
+											displayDateFormatter.formatDate(
+												dateRangePickerState.selectedEndDateMillis,
+												locale = Locale.getDefault()
+											)!!
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Just do `?: ""` instead

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/FavourFilter.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.components.switch.PrimarySwitch
+import org.uwaterloo.subletr.theme.filterRegularFont
+import org.uwaterloo.subletr.theme.secondaryButtonBackgroundColor
+import org.uwaterloo.subletr.theme.secondaryTextColor
+import org.uwaterloo.subletr.theme.subletrPink
+import org.uwaterloo.subletr.theme.textOnSubletrPink
+
+@Composable
+fun FavourFilter(
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/FavourFilter.kt]: Maybe we should just call it `FavouriteFilter`?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.pages.createlisting.DateInputButton
+import org.uwaterloo.subletr.pages.createlisting.DatePickerBottomSheet
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: These sorts of components that you're using from other screens can probably be moved into the `components` directory.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+		mutableStateOf(
+			if (currentDateRange.startingDate == null) ""
+			else parseUTCDateTimeToLocal(currentDateRange.startingDate!!)
+		)
+
+	}
+	var endButtonText by remember {
+		mutableStateOf(
+			if (currentDateRange.endingDate == null) ""
+			else parseUTCDateTimeToLocal(currentDateRange.endingDate!!)
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Do
```kotlin
currentDateRange.endingDate?.let {
	parseUTCDateTimeToLocal(it)
} ?: ""
```

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+fun DateFilter(
+	currentDateRange: HomeListUiState.DateRange,
+	updateDateFilter: (HomeListUiState.DateRange) -> Unit,
+	coroutineScope: CoroutineScope,
+	closeAction: () -> Unit,
+) {
+	var startButtonText by remember {
+		mutableStateOf(
+			if (currentDateRange.startingDate == null) ""
+			else parseUTCDateTimeToLocal(currentDateRange.startingDate!!)
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Do
```kotlin
currentDateRange.startingDate?.let {
	parseUTCDateTimeToLocal(it)
} ?: ""
```

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+							storeDateFormatISO
+						) else null,
+					),
+				)
+			closeAction()
+		},
+		revertInput = {
+			startButtonText =
+				if (currentDateRange.startingDate == null) ""
+				else parseUTCDateTimeToLocal(currentDateRange.startingDate!!)
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Do the same as above

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+			endButtonText =
+				if (currentDateRange.endingDate == null) ""
+				else parseUTCDateTimeToLocal(currentDateRange.endingDate!!)
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Do the same as above

File: android/app/src/main/java/org/uwaterloo/subletr/utils/DateUtils.kt [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+	val formatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME
+	val utcDateTime = LocalDateTime.parse(dateTimeString, formatter)
+
+	val utcZoneId = ZoneId.of("UTC")
+	val localZoneId = ZoneId.systemDefault()
+
+	val utcZonedDateTime = ZonedDateTime.of(utcDateTime, utcZoneId)
+	val localDateTime = utcZonedDateTime.withZoneSameInstant(localZoneId).toLocalDateTime()
+
+	return localDateTime.format(DateTimeFormatter.ofPattern("MM/dd/yyyy"))
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/utils/DateUtils.kt:19]: Add a localization TODO

File: android/app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -136,10 +136,6 @@
...
     <string name="roommate_preference">Roommate Preference</string>
     <string name="all_filters">All Filters</string>
-
-
-
-
-
-
+    <string name="dates">Dates</string>
+    <string name="favoured">Favoured</string>
```
Comment (leeminkyu16) [inline android/app/src/main/res/values/strings.xml]: I was thinking maybe this should be "Only show favourites" or something like that
Maybe wait for other opinions but I feel like Favoured is a odd word to use to relate with Favourites?

**Review by leeminkyu16 — APPROVED** [review]
Overall pretty good, just a few small things.
Let's Gamble, Try Merging

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt (hunk 4 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/DateFilter.kt]: Yeah, I was thinking about that. Will do.

**Review by andr-dev — APPROVED** [review]
Nice!

**Review by leeminkyu16 — APPROVED** [review]
Let's Gamble, Try Merging

**Pull request merged**